pkg/catalog/catalog.go

```diff
@@ -1110,7 +1110,7 @@ func (c *Catalog) ResetEntries(ctx context.Context, repositoryID string, branch
 	return c.Store.ResetPrefix(ctx, repository, branchID, keyPrefix, opts...)
 }
 
-func (c *Catalog) Commit(ctx context.Context, repositoryID, branch, message, committer string, metadata Metadata, date *int64, sourceMetarange *string, opts ...graveler.SetOptionsFunc) (*CommitLog, error) {
+func (c *Catalog) Commit(ctx context.Context, repositoryID, branch, message, committer string, metadata Metadata, date *int64, sourceMetarange *string, allowEmpty *bool, opts ...graveler.SetOptionsFunc) (*CommitLog, error) {
```

Comment:
Think we should refactor the signature - can be in a different pr and just pass commit params/options same struct layout as we pass to graveler so it will not stay a long list of arguments. When using struct the default value of each parameter will represent the default behaviour and the caller will not pass pointers to date and source metarange.